tools/c7n_gcp/c7n_gcp/resources/compute.py

```diff
@@ -64,6 +65,58 @@ def get_tag_value(self, instance):
         return instance.get('labels', {}).get(self.tag_key, False)
 
 
+@Instance.filter_registry.register('effective-firewall')
+class EffectiveFirewall(ValueFilter):
+    """Filters instances by their effective firewall rules.
+    See `getEffectiveFirewalls
+    <https://cloud.google.com/compute/docs/reference/rest/v1/instances/getEffectiveFirewalls>`_
+    for valid fields.
+
+    :example:
+
+    Filter all instances that have a firewall rule that allows public
+    acess
+
+    .. code-block:: yaml
+
+        policies:
+           - name: find-publicly-accessable-instances
+             resource: gcp.instance
+             filters:
+             - type: effective-firewall
+               key: firewalls[*].sourceRanges[]
+               op: in
+               value_type: swap
```

Comment:
fwiw, contains operator is a simpler way to express that can avoid the swap, plus allows use of value_type: cidr in this context.

Reply:
thanks good callout. will update